docs/opkssh-and-sssd.md

```diff
@@ -0,0 +1,125 @@
+# Use opkssh in combination with sssd
+
+In scenarions where the machines are already joined to a domain and where other `SSH` policies exist, `opkssh` login might run into issues.
+The SSH policies will take precedence based on the file name within `sshd_confi.d` and as such only the first matched policy will work.
+The following steps give an example on how to combine  2 different policies (SSSD and OPKSSH) so that one is used as a fallback, incase the other fails.
+
+## Scenario
+
+- `ssh` public keys of the users are stored in Active directory (AD)
+- `opkssh` installed on server and client
+- ssh server connected to domain via `SSSD`
+- only a specific group/groups are allowed to login via SSH to the servers
+
+## Implementation on Server
+
+- Add a custom policy under `/etc/ssh/sshd_config.d/50-ssh-domain-policy.conf` with the following contents.
+
+    ```
+
+    # Don't use authorized keys file from anyone.
+    AuthorizedKeysFile      none
+
+    # Don't trust user known host keys
+    IgnoreUserKnownHosts yes
+
+    # Rules for AD users
+    Match Group ssh-users
+    # Check if key exists in opkssh or in AD attribute "sshPublicKey"
+    AuthorizedKeysCommand /etc/ssh/sshd_config.d/parse_keys.sh %u %k %t
+    AuthorizedKeysCommandUser root
+    AuthenticationMethods publickey
+    ```
+    - The rule above basically restricts ssh to a specific subset of users and when these users try to log in a custom command is executed that acquires the user's ssh public key from the respective providers
+
+- Create another file under `/etc/ssh/sshd_config.d/parse_keys.sh` and set the rights to `700` with the following content
+
+    ```
+    #!/bin/bash
+
+    # Script to verify the ssh keys via both opkssh and sssd
+    # This is necessary to have a unified ssh policy that allows both
+
+
+    # Check keys via opkssh first
+    runuser -u opksshuser -- /usr/local/bin/opkssh verify $1 $2 $3
+
+    # If above errors then check via sssd
+
+    if [ "${PIPESTATUS[0]}" -ne 0 ]; then
+        runuser -u nobody --  /usr/bin/sss_ssh_authorizedkeys $1
+    fi
+    ```
+    - The script basically verifies with `opkssh` the user first and if the command fails then checks with `SSSD`. If both fail, then the login will be rejected. NOTE: `opkssh` requires multiple arguments, sssd requires only the username since the key is fethched from AD
+
+- For `opkssh` instead of specifying each user from the group, user a  policy plugin. Create the following files, `/etc/opk/policy.d/domain-plugin.yml` and `/etc/opk/policy.d/match-email.sh`
```

Comment:
```suggestion
- For `opkssh` instead of specifying each user from the group, use a  policy plugin. Create the following files, `/etc/opk/policy.d/domain-plugin.yml` and `/etc/opk/policy.d/match-email.sh`
```